clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/sagas/AbstractSagaAtomicOperation.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.orchestration.sagas;
+
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.netflix.spinnaker.clouddriver.orchestration.OrchestrationProcessor;
+import com.netflix.spinnaker.clouddriver.orchestration.SagaContextAware;
+import com.netflix.spinnaker.clouddriver.orchestration.sagas.SagaAtomicOperationBridge.ApplyCommandWrapper;
+import com.netflix.spinnaker.clouddriver.orchestration.sagas.SagaAtomicOperationBridge.ApplyCommandWrapper.ApplyCommandWrapperBuilder;
+import com.netflix.spinnaker.clouddriver.saga.SagaService;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaFlow;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Removes some of the boilerplate for AtomicOperations to use Sagas.
+ *
+ * @param <T> The AtomicOperation description
+ * @param <R> The operation result type
+ */
+public abstract class AbstractSagaAtomicOperation<T, R>
```

Comment:
My first thought when reading this is - why can't we use this for `DeployHandler` implementations too?  It's a similar process - we create a saga flow and then return a result via `SagaAtomicOperationBridge`.  I think if we make this work with for both cases, that would be better.